### DISCUSSION TITLE
Trusted proxies in docker behind Traefik and through a VPN gateway

### DISCUSSION THREAD

**Discussion by dror3go** [photoprism/photoprism#3516] [Q&A]
My PhotoPrism is deployed in a Docker container behind Traefik as a reverse proxy, and the host is connected to a VPN gateway: external machine with a static IP to where the DNS is pointing and where a WireGuard server is installed. My server is acting as a client connected to the VPN.

What are the trusted proxies that I need to configure in `PHOTOPRISM_TRUSTED_PROXY` in order to see the real IPs under the sessions management page? Does the order matter?

**Comment by lastzero** [comment]
The trusted proxy value is a string containing the CIDR ranges or IPv4/v6 addresses from which reverse proxy headers can be trusted, separated by commas. Since most users use our software with Docker, the default value is the internal IP range used by Docker: `"172.16.0.0/12"`

- In your particular case, the trusted proxy would probably be Traefik.
- Whether you need to set a custom IP address or range depends on your local network configuration.
- For example, your proxy might be running directly on the host or on another server. In this case, the internal Docker IP range will not work for you.

Without knowing all the details, I unfortunately cannot give you a more precise answer.

**Comment by dror3go** [comment]
Traefik is running in a Docker container on the same host much like PhotoPrism itself.
I guess I need a small tweak to make it work, but not sure what it is exactly. Would appreciate a pointer here :pray:

**Comment by doebi** [comment]
If both of your machines are in your wireguard VPN, there is no need of traefik and external IP at all. I am running photoprism on one of my home servers in a wireguard (spec. tailscale) VPN and simply connect to `hostname:2342`

That way its even more protected from unauthenticated access by others.

**Reply by dror3go** [reply]
What do you mean by "both machines" and how is that related to my question? I'm exposing my inatance to the internet, while you seem to expose it only in your private network.

**Reply by doebi** [reply]
From your inital question it wasn't obvious to me how your intended setup looks like. I just pointed out how I use it.
Both machines meaning: client (my notebook) and server (a remote machine running photoprism)

As @lastzero pointed out, the `PHOTOPRISM_TRUSTED_PROXY` setting refers to the IP range
Your issue is most probably traefik or docker related, but without any further information it is hard to diagnose.

From my experience: I had to set up multiple (docker-)networks and carefully specifiy routing between them to make similar setups work.
